--- lib/librte_eal/linuxapp/igb_uio/igb_uio.c
+++ lib/librte_eal/linuxapp/igb_uio/igb_uio.c
@@ -388,14 +388,11 @@
 		}
 		/* fall back to INTX */
 	case RTE_INTR_MODE_LEGACY:
-		if (pci_intx_mask_supported(dev)) {
-			dev_dbg(&dev->dev, "using INTX");
-			udev->info.irq_flags = IRQF_SHARED;
-			udev->info.irq = dev->irq;
-			udev->mode = RTE_INTR_MODE_LEGACY;
-			break;
-		}
-		dev_notice(&dev->dev, "PCI INTX mask not supported\n");
+		dev_dbg(&dev->dev, "using INTX");
+		udev->info.irq_flags = IRQF_SHARED;
+		udev->info.irq = dev->irq;
+		udev->mode = RTE_INTR_MODE_LEGACY;
+		break;
 		/* fall back to no IRQ */
 	case RTE_INTR_MODE_NONE:
 		udev->mode = RTE_INTR_MODE_NONE;
