diff -Naur pktgen-3.1.2/app/Makefile pktgen-3.1.2-fixed/app/Makefile
--- pktgen-3.1.2/app/Makefile   2017-02-06 17:09:24.000000000 -0700
+++ pktgen-3.1.2-fixed/app/Makefile      2017-06-27 03:53:01.483171669 -0600
@@ -66,6 +66,7 @@

 MYLIB_PATH := -L$(LUA_LIB) -L$(COMMON_LIB)
 MYLIBS := -llua -lcommon
+LDLIBS_PATH := $(LUA_LIB) $(COMMON_LIB)

 ifeq ($(CONFIG_RTE_BUILD_SHARED_LIB),y)
        MYLIBS += -lrte_pmd_ixgbe

