diff -Naur pktgen-v3.1.0/app/Makefile pktgen-v3.1.0-fixed/app/Makefile
--- pktgen-v3.1.0/app/Makefile	2016-12-09 16:43:44.000000000 -0600
+++ pktgen-v3.1.0-fixed/app/Makefile	2017-06-15 15:07:37.528358051 -0500
@@ -66,6 +66,7 @@
 
 MYLIB_PATH := -L$(LUA_LIB) -L$(COMMON_LIB)
 MYLIBS := -llua -lcommon
+LDLIBS_PATH := $(LUA_LIB) $(COMMON_LIB)
 
 ifeq ($(CONFIG_RTE_BUILD_SHARED_LIB),y)
 	MYLIBS += -lrte_pmd_ixgbe
